### DISCUSSION TITLE
Include 'altText' in GraphQL queries

### DISCUSSION THREAD

**Discussion by Janek9527** [sanity-io/sanity-plugin-media#30] [Q&A]
Hello,
first of all - thanks for this plugin!

Is it possible to include fields like 'altText' in GraphQL queries when using sanitys 'sanity graphql deploy' ?
I installed the plugin following the instructions and it is working well and i can also query the fields using GROQ.

If it is possible, could you please give me instructions how to set this up correctly using GraphQL?

Thanks and best regards
Janek

**Comment by robinpyon** [comment]
Hey @Janek9527, thanks for giving the plugin a go!

I'm not super familiar with GraphQL so please take the below with a grain of salt

I don't believe this is currently possible with Sanity's GraphQL endpoint (with `sanity graphql deploy`) since it generates its schema based off your user-defined studio schema.

This plugin exposes a custom schema (for tags) – which their GraphQL endpoint does correctly pick up – but it's not possible to redefine / augment `sanity.imageAsset` and `sanity.fileAsset` and expose custom fields as they are locked system types. 

AFAIK, for this to work, Sanity would need to be able to provide a way for custom asset source plugins (such as this one) to formally define custom fields which would then be added to the `sanity.imageAsset` and `sanity.fileAsset` schemas respectively. Or alternatively, allow user-defined custom fragments which can be used place of their default types.

However – if you are using Gatsby and aren't bound to using Sanity's GraphQL endpoint, it might be worth looking into [gatsby-plugin-sanity-image](https://github.com/coreyward/gatsby-plugin-sanity-image) which does in fact let you define custom fragments to be used in place of the `SanityImage` type (See the `fragmentTypeName` option [here](https://github.com/coreyward/gatsby-plugin-sanity-image#configuration-directives)).
